documentation/book/con-the-upgrade-process.adoc

```diff
@@ -0,0 +1,62 @@
+// This module is included in the following assemblies:
+//
+// assembly-upgrading-kafka-versions.adoc
+
+[id='con-the-upgrade-process-{context}']
+= The upgrade process
+
+A given version of the Cluster Operator knows certain details about a given set of Kafka versions. 
+For the purpose of a Kafka upgrade the important attributes are:
+
+* The version of the Kafka protocol to be used between brokers. (this is set via the `interbroker.protocol.version` Kafka configuration parameter),
+* The format version to be used for messages stored in the Kafka log (this is set via the `log.message.format.version` Kafka configuration parameter),
+* The required version of Zookeeper.
+
+The differences in these attributes between the Kafka version being upgraded from and the Kafka version being upgraded to will determine exactly how the upgrade process works.
+
+== When all the attributes are the same
```

Comment:
I feel like these following headings could be put into a section table.  Something like this: http://enmasse.io/documentation/master/openshift/#ref-console-address-filtering-messaging 

Where we have 2 columns.  In Column 1 we say 'When all the attributes are the same', in Column 2 we say 'Do This'

Column 2 can then be more succinct like "Update `Kafka.spec.kafka.config` with the version"

Reply:
I didn't review much further, because I'd like to hear what others think about this.  Right now I'm finding this section a bit clunky.

Reply:
I'm having trouble figuring out how to see the previous version of the file.  Any tips?

Reply:
https://github.com/strimzi/strimzi-kafka-operator/blob/bd5469aaf9a73ada3ac6e939e59a5af582477950/documentation/book/con-the-upgrade-process.adoc